homeassistant/components/mobile_app/webhook.py

```diff
@@ -172,3 +182,71 @@
 
         return webhook_response(safe_registration(new_registration),
                                 registration=registration, headers=headers)
+
+    if webhook_type == WEBHOOK_TYPE_REGISTER_SENSOR:
+        entity_type = data[ATTR_SENSOR_TYPE]
+
+        unique_id = data[ATTR_SENSOR_UNIQUE_ID]
+
+        unique_store_key = "{}_{}".format(webhook_id, unique_id)
+
+        if unique_store_key in hass.data[DOMAIN][entity_type]:
+            _LOGGER.error("Refusing to re-register existing sensor %s!",
+                          unique_id)
+            return empty_okay_response()
+
+        data[CONF_WEBHOOK_ID] = webhook_id
+
+        hass.data[DOMAIN][entity_type][unique_store_key] = data
+
+        try:
+            await hass.data[DOMAIN][DATA_STORE].async_save(savable_state(hass))
+        except HomeAssistantError as ex:
+            _LOGGER.error("Error updating mobile_app registration: %s", ex)
+            return empty_okay_response()
+
+        register_signal = '{}_{}_register'.format(DOMAIN,
+                                                  data[ATTR_SENSOR_TYPE])
+        async_dispatcher_send(hass, register_signal, data)
+
+        return webhook_response({"status": "registered"},
+                                registration=registration, headers=headers)
```

Comment:
Why would we return registration ? We should never return the full registration, there is so much info we don't need to expose. Including secrets that don't have to go over the wire.

Reply:
oh, it's not part of the response, oops.

Reply:
`webhook_response` doesn't return the registration, it just uses it to determine if we need to send back an encrypted payload.